benchmarks/symbench.m

```diff
@@ -0,0 +1,64 @@
+Remove["Global`*"]
+ClearAll["Global`*"]
+(*define f*)
+f[z_]=z*z/3^(1/2)+1/3;
+
+(*define R1*)
+h=1/2;
+"Time for R1 : " 
+R1:=(Expand[f[f[f[f[f[f[f[f[f[f[h]]]]]]]]]]];);
+Print[AbsoluteTiming[R1]*1000];
+Remove["Global`*"]
+ClearAll["Global`*"]
+
+(*define hermite*)
+hermite[1,y_]=y/2;
+hermite[0,y_]=1;
+hermite[n_,y_]:=Expand[2*y*hermite[n-1,y]-2*(n-1)*hermite[n-2,y]];
+
+(*define R2*)
+R2[n_,y_]:=hermite[n,y];
+n=15;
+"Time for R2 : " 
+Print[1000*AbsoluteTiming[R2[n,y];]];
+Remove["Global`*"]
+ClearAll["Global`*"]
+
+(*define R3*)
+f[]=x+y+z;
+Array[fc, 10];
+"Time for R3: " 
+Print[AbsoluteTiming[Do[fc[i] =Boole[(f[]==f[])],{i,1,10}]]*1000]
+
+(*define R7*)
+"Time for R7: " 
+f[x_]=x^24+34* x^12+45*x^3+9* x^18+34* x^10+32*x^21;
```

Comment:
This does not involve symbolics. What you should do is to create the expression first and then measure the time taken to replace `x` with `0.5`